src/Features/Core/Portable/QuickInfo/CommonSemanticQuickInfoProvider.cs

```diff
@@ -296,6 +304,91 @@ void AddSection(string kind, ImmutableArray<TaggedText> taggedParts)
             return default;
         }
 
+        private static ImmutableArray<TaggedText> GetConstantValueContent(
+            SyntaxToken token,
+            SemanticModel semanticModel,
+            ISyntaxFactsService syntaxFacts,
+            ISymbolDisplayService displayService,
+            CancellationToken cancellationToken)
+        {
+            if (!syntaxFacts.IsBinaryExpression(token.Parent))
+            {
+                return default;
+            }
+
+            syntaxFacts.GetPartsOfBinaryExpression(token.Parent, out var leftExpression, out var operatorToken, out var rightExpression);
+
+            if (token != operatorToken)
+            {
+                return default;
+            }
+
+            var constant = semanticModel.GetConstantValue(token.Parent, cancellationToken);
+            if (!constant.HasValue)
+            {
+                return default;
+            }
+
+            var leftConstant = semanticModel.GetConstantValue(leftExpression, cancellationToken);
+            if (!leftConstant.HasValue)
+            {
+                return default;
+            }
+
+            var rightConstant = semanticModel.GetConstantValue(rightExpression, cancellationToken);
+            if (!rightConstant.HasValue)
+            {
+                return default;
+            }
+
+            var textBuilder = ImmutableArray.CreateBuilder<TaggedText>();
```

Comment:
i still would not do this.

Reply:
Would not do what? Use an array builder?

Reply:
i would not piece together an expression as the result.  I would just take the final constant value for the binary expressoin and show that.  As a bonus, if you are dealing with expressions with large constants,there's less that needs to be trimmed.

Reply:
Ask @sharwell about that, I am not in charge of the design.